ros_buildfarm/templates/release/trigger_upload_repo_job.xml.em

```diff
@@ -0,0 +1,42 @@
+<project>
+  <actions/>
+  <description/>
+  <keepDependencies>false</keepDependencies>
+  <properties>
+    <jenkins.advancedqueue.priority.strategy.PriorityJobProperty plugin="PrioritySorter@@3.5.0">
+      <useJobPriority>false</useJobPriority>
+      <priority>-1</priority>
+    </jenkins.advancedqueue.priority.strategy.PriorityJobProperty>
+    <org.jenkinsci.plugins.requeuejob.RequeueJobProperty plugin="jobrequeue@@1.1">
+      <requeueJob>false</requeueJob>
+    </org.jenkinsci.plugins.requeuejob.RequeueJobProperty>
+  </properties>
+  <scm class="hudson.scm.NullSCM"/>
+  <assignedNode>building_repository</assignedNode>
+  <canRoam>false</canRoam>
+  <disabled>false</disabled>
+  <blockBuildWhenDownstreamBuilding>false</blockBuildWhenDownstreamBuilding>
+  <blockBuildWhenUpstreamBuilding>true</blockBuildWhenUpstreamBuilding>
```

Comment:
This seems to be `false` for the existing `upload_testing` job deployed on Jenkins. The question is is there reason for that on Jenkins?

Reply:
Because the testing upload is triggered by the sync which is triggered by the builds automatically if this waited for upstream it could only upload to testing when the farm had gone completely idle.

For the upload_main it's ok to have it block since it's only one level of dependency on the manually triggered sync-to-main jobs. So it will stop it from running if there's any sync-to-main job queued. This makes it a little safer if two sync-to-main-jobs get triggered close together, you'll get the aggregate uploaded instead of potentially 2 uploads. It should be safe to do it either way as they're all locked to the same executor. I was mostly looking to avoid the number of times that the public mirror needed to sync since we have seen issues with that and have little to no introspection into it's state.

Reply:
Addressing this as well.